assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,43 @@
-console.log('***** Music Collection *****')
+console.log("***** Music Collection *****");
+collection = [];
+
+function addToCollection(title, artist, yearPublished) {
+  let tempCollection = {
+    title: `${title}`,
+    artist: `${artist}`,
+    yearPublished: `${yearPublished}`,
+  };
+  collection.push(tempCollection);
+  console.log(collection);
+}
+console.log(collection);
+
+addToCollection("Little Sunshine", "Ghost", 2022);
+addToCollection("Holy Roller", "Spirit Box", 2020);
+addToCollection("Pices", "Jinjer", 2016);
+addToCollection("Cold Like War", "We Came As Romans", 2017);
+addToCollection("Voices in My Head", "Falling in Reverse", 2022);
+addToCollection("Toxic", "Lauren Babic", 2019);
+
+function showCollection(arrayParam) {
+  console.log(arrayParam.length);
+  for (let songs of arrayParam) {
+    console.log(
+      `${songs.title} by ${songs.artist} published in ${songs.yearPublished}`
+    );
+  }
+}
+showCollection(collection);
+
+//
+function findByArtist(artistString) {
+  artistArray = [];
+  for (let i = 0; i < collection.length; i++) {
+    if (collection[i].artist === artistString) {
+      artistArray.push(collection[i]);
+      console.log(artistArray);
```

Comment:
Once again, the requirements call for `findByArtist` to _return_ the results of the search, not log them. Since this function only has a console.log, no return, it is useless for the rest of your code. All it can do is print stuff out for the developer to read.

Reply:
Also worth noting that you put the console.log _inside_ the loop, which may have been a mistake. Every time you find a match you are going to log out the whole array. That is pretty redundant. The return (or console.log if you really wanted to use one), would make more sense after loop is done and the `artistArray` is complete.